src/back/models/Project.ts

```diff
@@ -15,8 +16,27 @@ export type ProjectAttributes = {
   created_at: Date
 }
 
+// TODO: add here all data from other tables (updates, personnel, milestones, etc)
+export type ProjectAugmented = ProjectAttributes & {
+  author: string
+  coauthors: string[] | null
+}
+
 export default class ProjectModel extends Model<ProjectAttributes> {
   static tableName = 'projects'
   static withTimestamps = false
   static primaryKey = 'id'
+
+  static async getProject(id: string) {
+    const query = SQL`
+    SELECT pr.id, pr.proposal_id, p.user as author, array_agg(co.address) FILTER (WHERE co.status = 'APPROVED') as coauthors, pr.title, pr.status, pr.links, pr.about, pr.about_updated_by, pr.about_updated_at, pr.updated_at, pr.created_at
```

Comment:
Nice! Since we are using all project data I think you could just `SELECT pr.*`

Also, we won't use any unapproved coauthors data in the query, so maybe you could just filter them in the `ON` clause of the join, before aggregating.

```sql
SELECT pr.*, p.user as author, array_agg(co.address) as coauthors
FROM projects pr
JOIN proposals p on pr.proposal_id = p.id
LEFT JOIN coauthors co ON pr.proposal_id = co.proposal_id AND co.status = 'APPROVED'
WHERE pr.id = ${id}
GROUP BY pr.id;
```